library/src/main/java/com/datatorrent/lib/join/POJOInnerJoinOperator.java

```diff
@@ -0,0 +1,247 @@
+          inputFieldObjects[i].timeFieldGet = PojoUtils.createGetter(inputClass, timeFields.get(i), tc);
+        }
+        for (int j = 0; j < includeFields[i].length; j++) {
+          Class ic = ClassUtils.primitiveToWrapper(inputClass.getField(includeFields[i][j]).getType());
+          Class oc = ClassUtils.primitiveToWrapper(outputClass.getField(includeFields[i][j]).getType());
+          inputFieldObjects[i].fieldMap.put(PojoUtils.createGetter(inputClass, includeFields[i][j], ic),
+              PojoUtils.createSetter(outputClass, includeFields[i][j], oc));
+        }
+      } catch (NoSuchFieldException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  /**
+   * Extract the key value from the given tuple
+   * @param tuple given tuple
+   * @param isStream1Data Specifies whether the given tuple belongs to stream1 or not.
+   * @return
+   */
+  @Override
+  public Object extractKey(Object tuple, boolean isStream1Data)
+  {
+    return isStream1Data ? inputFieldObjects[0].keyGet.get(tuple) :
+      inputFieldObjects[1].keyGet.get(tuple);
+  }
+
+  /**
+   * Merge the given tuples
+   * @param tuple1 tuple belongs to stream1
+   * @param tuple2 tuple belongs to stream1
+   * @return
+   */
+  @Override
+  public Object mergeTuples(Object tuple1, Object tuple2)
+  {
+    Object o;
+    try {
+      o = outputClass.newInstance();
+      for (Map.Entry<PojoUtils.Getter,PojoUtils.Setter> g: inputFieldObjects[0].fieldMap.entrySet()) {
+        g.getValue().set(o, g.getKey().get(tuple1));
+      }
+      for (Map.Entry<PojoUtils.Getter,PojoUtils.Setter> g: inputFieldObjects[1].fieldMap.entrySet()) {
+        g.getValue().set(o, g.getKey().get(tuple2));
+      }
+    } catch (InstantiationException | IllegalAccessException e) {
+      throw new RuntimeException(e);
+    }
+    return o;
+  }
+
+  /**
+   * Emit the given tuple through the outputPort
+   * @param tuple given tuple
+   */
+  @Override
+  public void emitTuple(Object tuple)
+  {
+    outputPort.emit(tuple);
+  }
+
+  @Override
+  public void activate(Context context)
+  {
+    generateSettersAndGetters();
+  }
+
+  @Override
+  public void deactivate()
+  {
+  }
+
+  @Override
+  public void endWindow()
+  {
+    super.endWindow();
+    time += timeIncrement;
+  }
+
+  /**
+   * Returns the streamcodec for the streams
+   * @param isStream1data Specifies whether the codec needs for stream1 or stream2.
+   * @return the object of InnerJoinStreamCodec
+   */
+  public StreamCodec<Object> getInnerJoinStreamCodec(boolean isStream1data)
+  {
+    if (isStream1data) {
+      return new InnerJoinStreamCodec(getKeyFieldsStr().split(",")[0]);
+    }
+    return new InnerJoinStreamCodec(getKeyFieldsStr().split(",")[1]);
+  }
+
+  private class FieldObjectMap
+  {
+    public Class<?> inputClass;
+    public PojoUtils.Getter keyGet;
+    public PojoUtils.Getter timeFieldGet;
+    public Map<PojoUtils.Getter,PojoUtils.Setter> fieldMap;
```

Comment:
If it has a single mapping then it's difficult to check whether the getter is relates to which of the stream.